exist-core/src/main/java/org/exist/http/RESTServer.java

```diff
@@ -1162,61 +1191,66 @@ public void doDelete(final DBBroker broker, final Txn transaction, final String
 
     private boolean checkForXQueryTarget(final DBBroker broker, final Txn transaction,
         final XmldbURI path, final HttpServletRequest request,
-        final HttpServletResponse response) throws PermissionDeniedException,
-        NotFoundException, IOException, BadRequestException {
-        
+        final HttpServletResponse response) throws PermissionDeniedException, IOException, BadRequestException {
+
         if (request.getAttribute(XQueryURLRewrite.RQ_ATTR) == null) {
             return false;
         }
         final String xqueryType = MimeType.XQUERY_TYPE.getName();
+
         final Collection collection = broker.getCollection(path);
-        if (collection == null) {
-            XmldbURI servletPath = path;
-            LockedDocument lockedDocument = null;
-            DocumentImpl resource = null;
-            // work up the url path to find an
-            // xquery resource
-            while (null == resource) {
-                // traverse up the path looking for xquery objects
+        // a collection is not executable
```

Comment:
executable means "has executable flag' ? = user has access to Collection ?

Reply:
No, here executable means "is an xquery".

Reply:
confusing still...